core/src/main/java/org/eclipse/hono/util/CredentialsObject.java

```diff
@@ -0,0 +1,81 @@
+/**
+ * Copyright (c) 2017 Bosch Software Innovations GmbH.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *    Bosch Software Innovations GmbH - initial creation
+ */
+package org.eclipse.hono.util;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Encapsulates the credentials information for a device that was found by the get operation of the
+ * <a href="https://www.eclipse.org/hono/api/Credentials-API/">Credentials API</a>.
+ * <p>
+ * Is mapped internally from json representation by jackson-databind.
+ */
+public final class CredentialsObject {
+    @JsonProperty(CredentialsConstants.FIELD_DEVICE_ID)
+    private String deviceId;
+    @JsonProperty(CredentialsConstants.FIELD_TYPE)
+    private String type;
+    @JsonProperty(CredentialsConstants.FIELD_AUTH_ID)
+    private String authId;
+    @JsonProperty(CredentialsConstants.FIELD_ENABLED)
+    private Boolean enabled;
+    /*
+     * Since the format of the secrets field is not determined by the Credentials API, they are best represented as
+     * key-value maps with key and value both of type String.
+     * The further processing of secrets is part of the validator for the specific type.
+     */
+    @JsonProperty(CredentialsConstants.FIELD_SECRETS)
+    private List<Map<String,String>> secrets;
```

Comment:
This kind of invalidates the whole idea of mapping the JSON to a concrete class FMPOV. In the end, this is the interesting information and now validators need to fiddle around with a map of strings to strings, i.e. they need to parse the string into something useful, handle all errors etc.
It almost looks like doing the mapping wasn't a good idea in the first place. Before we had a consistent API (JsonObject) to use for interacting with JSON payloads. Now we have property based access for trivial stuff (e.g. *deviceId*) and no support at all for accessing the interesting (and more complex) parts :-(
In addition, you had to make changes in many other places as well in order to support `CredentialsObject` in addition to `JsonObject`. All in all, I do not think that this is very helpful but instead makes the code less understandable by introducing the distinction between CredentialsObject and all other JSON payload.